### PR TITLE
[CHK-919] feat: add transaction origin header apim policy

### PR DESCRIPTION
**Pull request by pietro-tota**
### List of changes
Add APIM policy to set transaction origin `x-transaction-origin` for POST transactions API

### Motivation and context
This modification is needed in order to track transaction origin for new transactions

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/core/apim_checkout.tf [hunk 1]
```diff
@@ -348,6 +348,15 @@ resource "azurerm_api_management_api_operation_policy" "transaction_authorizatio
   )
 }
 
+resource "azurerm_api_management_api_operation_policy" "new_transaction_origin" {
```
Comment (infantesimone) [inline src/core/apim_checkout.tf:351]: this policy should be extended to all ecommerce API for checkout

Comment (pietro-tota) [reply src/core/apim_checkout.tf:351]: So can I move this policy to the base one?

Comment (infantesimone) [reply src/core/apim_checkout.tf:351]: You can delete `azurerm_api_management_api_operation_policy.new_transaction_origin` and add 
```xml
    <set-header name="X-Client-Id" exists-action="override" >
      <value>CHECKOUT</value>
    </set-header>
```
in
`src/domains/ecommerce-app/api/ecommerce-transactions-service/v1/_base_policy.xml.tpl`

File: src/core/api/checkout/checkout_ecommerce/v1/_openapi.json.tpl [hunk 2]
```diff
@@ -111,6 +111,18 @@
     "/transactions": {
       "post": {
         "operationId": "newTransaction",
+        "parameters": [
```
Comment (infantesimone) [inline src/core/api/checkout/checkout_ecommerce/v1/_openapi.json.tpl:114]: To delete, this is parameter of `pagopa-ecommerce-transaction-service` and not of `ecommerce-checkout` API. So, the client must not set this header.

File: src/core/apim_checkout.tf (hunk 1 above)
Comment (pietro-tota) [reply src/core/apim_checkout.tf]: fixed in https://github.com/pagopa/pagopa-infra/pull/607/commits/6e372723ccde22af54cf8182cc3ed44bac0138ab

File: src/core/api/checkout/checkout_ecommerce/v1/_openapi.json.tpl (hunk 2 above)
Comment (pietro-tota) [reply src/core/api/checkout/checkout_ecommerce/v1/_openapi.json.tpl]: fixed in https://github.com/pagopa/pagopa-infra/pull/607/commits/6e372723ccde22af54cf8182cc3ed44bac0138ab

**Review by infantesimone — APPROVED** [review]

**Pull request merged**